main.go

```diff
@@ -112,17 +116,36 @@ func readStdin() string {
 	return b.String()
 }
 
+// defaultConfig returns the default config path
+func defaultConfig() string {
+	if c := os.Getenv("NAP_CONFIG"); c != "" {
+		return c
+	}
+	cfgPath, err := xdg.ConfigFile("nap/config.yaml")
+	if err != nil {
+		return "config.yaml"
+	}
+	return cfgPath
+}
+
 // readConfig returns a configuration read from the environment.
 func readConfig() Config {
-	config := Config{Home: defaultHome()}
-	if err := env.Parse(&config); err != nil {
-		return Config{
-			Home:            defaultHome(),
-			File:            defaultSnippetFileName,
-			Theme:           "dracula",
-			DefaultLanguage: "go",
+	config := newConfig()
+	fi, err := os.Open(defaultConfig())
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return newConfig()
+	}
+	if fi != nil {
+		defer fi.Close()
+		if err := yaml.NewDecoder(fi).Decode(&config); err != nil {
+			return newConfig()
 		}
 	}
+
+	if err := env.Parse(&config); err != nil {
```

Comment:
I've removed the `envDefault` struct tags, as they were causing this to overwrite config file values even if the env wasn't defined. Now, rather, a config is just pre-created with default values and then overwritten as things unmarshal onto it.